src/coreclr/src/jit/hwintrinsicxarch.cpp

```diff
@@ -747,6 +748,70 @@ GenTree* Compiler::impBaseIntrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
+        case NI_Vector128_Create:
+        case NI_Vector256_Create:
+        {
+#if defined(TARGET_X86)
+            if (varTypeIsLong(baseType))
+            {
+                // TODO-XARCH-CQ: It may be beneficial to emit the movq
+                // instruction, which takes a 64-bit memory address and
+                // works on 32-bit x86 systems.
+                break;
```

Comment:
It's not something we are handling today anywhere else and we have an identical `TODO` in the other places.

I believe this would just require us creating a `GT_IND` node since x86 requires it be a `long*` or `ulong*`, but I'm not positive if that is the ideal/correct way to handle it

Reply:
Just noting, this is also not something we are handling in the managed implementation today.

Reply:
Right, I gathered that, but was just curious, given the comment.

Reply:
I'm not super familiar with the differences between the different addressing forms in the JIT and when each is appropriate to use.

Do we have a helper function that will create a `T*` from an arbitrary `T` (In this case a `ulong*` from a `ulong`) when the given operand on the stack could be a constant, local, field, byref, or other indirection?

Reply:
IIUC you really want a `GT_ADDR`, to get the address of an operand. I the JIT addresses are not strongly typed, they are always `TYP_BYREF`, so it would just be something like:
```
GenTree* addr = gtNewOperNode(GT_ADDR, TYP_BYREF, op);
```

Reply:
Though I'd note that I'm not really pressing for you to do this now - I was just curious why you added the comment rather than doing it.

Reply:
I'll attempt this in a follow up PR, hopefully it is that straightforward 😄